src/sql/base/common/locConstants.ts

```diff
@@ -52,3 +52,6 @@ export const invalidArgumentsError = localize('vscodeInvalidArgumentsError', "In
 export const docCreationFailedError = localize('vscodeDocCreationFailedError', "Failed to create notebook document.");
 export const cellToolbarCompatibilityMessage = localize('notebook.cellToolbarLocation.compatibilityDescription', "Where the cell toolbar should be shown, or whether it should be hidden. Note: This setting is only enabled for extension compatibility purposes, and so does not affect anything.");
 export const docNotFoundForUriError = localize('docNotFoundForUriError', 'Could not open a notebook document for the specified URI.');
+export function versionSyntax(engine: string, version: string): string { return localize('sql.versionSyntax', "Could not parse `{0}` value {1}. Please use, for example: ^1.22.0, ^1.22.x, etc.", engine, version) }
+export function versionMismatch(currentVersion: string, requestedVersion: string): string { return localize('sql.versionMismatch', "Extension is not compatible with Azure Data Studio {0}. Extension requires: {1}.", currentVersion, requestedVersion); }
+export function versionMismatchVsCode(currentVersion: string, requestedVersion: string, supportedVersion: string): string { return localize('sql.versionMismatchVsCode', "Extension is not compatible with Azure Data Studio {0}. Extension requires a newer VS Code Engine Version {1}, which is newer than what is currently supported ({2}).", currentVersion, requestedVersion, supportedVersion); }
```

Comment:
I added a separate specialized message here to more clearly indicate when it fails because of the azdata engine versus the vs code engine being mismatched.